int64 row
int64 column
float64 x
float64 y
float64 best_angle
float64[] probabilities